lumenlearning/lti-middleware
# L3-93 Switched from session storage to cookie storage

## Description
I switched from session storage to cookie storage for the state and nonce used to do the LTI Core authentication process. I also bumped the Spring version in light of the latest vulnerability (though this is simply precautionary as I believe either Spring MVC or Spring WebFlux is used in this project).

### Motivation and Context
Switching from session storage to cookie storage will hopefully allow for having multiple instances of the middleware running without having to utilize sticky sessions. By storing the state and nonce in cookies in the user's browser, they remain tied to the user's browser without being tied to the session storage of a specific instance of the middleware.

### Fixes
[L3-93](https://lumenlearning.atlassian.net/browse/L3-93)

## How Has This Been Tested?
1. Run the middleware with demo mode off.
2. Log into Canvas and click on a Waymaker link with the developer tools open.
3. Note that the link works as expected and that the LTI 1.3 handshake took place in the developer tools (/oidc/login_initiations and /lti3 endpoints were successfully hit).
4. Note that there are no unexpected errors in the middleware logs.
5. Run the middleware with demo mode on.
6. Log into Canvas and click on a Waymaker link with the developer tools open.
7. When it stops after the /oidc/login_initiations request, alter or delete the lti_state cookie.
8. Click to go on to the second step.
9. Note that the error page displays and that there is an error in the logs.
10. Click on a Waymaker link with the developer tools open again.
11. When it stops after the /oidc/login_initiations request, alter or delete the lti_nonce cookie.
12. Click to go on to the next step.
13. Note that the error page displays and that there is an error in the logs.

---

## Checklist
